src/logger/httpLoggerRules_test.go

```diff
@@ -286,195 +286,662 @@ func TestUsesRemoveRules(t *testing.T) {
-
-    queue = make([]string, 0)
-    logger = NewHttpLoggerQueueRules(queue, "!request_body! remove_if !.*!")
-    httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
-    assert.Equal(t, 1, len(queue), "queue length is not 1")
-    assert.Equal(t, false, strings.Contains(queue[0], "[\"request_body\","), "request_body not removed")
-    assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\","), "response_body not found")
-
-    queue = make([]string, 0)
-    logger = NewHttpLoggerQueueRules(queue, "!response_body! remove_if !.*!")
-    httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
-    assert.Equal(t, 1, len(queue), "queue length is not 1")
-    assert.Equal(t, true, strings.Contains(queue[0], "[\"request_body\","), "request_body not found")
-    assert.Equal(t, false, strings.Contains(queue[0], "[\"response_body\","), "response_body not removed")
-
-    queue = make([]string, 0)
-    logger = NewHttpLoggerQueueRules(queue, "!response_body|request_body! remove_if !.*World.*!") //mock response should contain "World" and therefore be removed
-    httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
-    assert.Equal(t, 1, len(queue), "queue length is not 1")
-    assert.Equal(t, true, strings.Contains(queue[0], "[\"request_body\","), "request_body not found")
-    assert.Equal(t, false, strings.Contains(queue[0], "[\"response_body\","), "response_body not removed")
-
-    queue = make([]string, 0)
-    logger = NewHttpLoggerQueueRules(queue, "!response_body|request_body! remove_if !.*blahblahblah.*!")
-    httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
-    assert.Equal(t, 1, len(queue), "queue length is not 1")
-    assert.Equal(t, true, strings.Contains(queue[0], "[\"request_body\","), "request_body not found")
-    assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\","), "response_body not found")
-
-    queue = make([]string, 0)
-    logger = NewHttpLoggerQueueRules(queue, "!request_body! remove_if !.*!\n!response_body! remove_if !.*!")
-    httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
-    assert.Equal(t, 1, len(queue), "queue length is not 1")
-    assert.Equal(t, false, strings.Contains(queue[0], "[\"request_body\","), "request_body not removed")
-    assert.Equal(t, false, strings.Contains(queue[0], "[\"response_body\","), "response_body not removed")
+	helper := GetTestHelper()
+
+	request := helper.MockRequestWithJson2()
+	mockResponse := helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml))
+	mockResponse.Request = request
+
+	queue := make([]string, 0)
+	logger := NewHttpLoggerQueueRules(queue, "!response_header:blahblahblah! remove_if !.*!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
```

Comment:
Unlike the java logger httpmessage is not a class or struct.... the `send...` method is known throughout the `package logger`